src/openrct2/actions/StaffHireNewAction.cpp

```diff
@@ -190,6 +190,8 @@ GameActions::Result StaffHireNewAction::QueryExecute(bool execute) const
 
             // INVESTIGATE: x and y are LOCATION_NULL at this point.
             newPeep->MoveTo(newPeep->GetLocation());
+            network_set_pickup_peep(network_get_current_player_id(), newPeep);
```

Comment:
I'm not sure this will work. This action will be getting called by networked clients so you'll end up setting a peep to the wrong player id.

Reply:
Thanks @duncanspumpkin. I did some testing and this indeed does not work in online games. Will take some time to review alternative options.